komfydb/execution/seq_scan.cc

```diff
@@ -0,0 +1,55 @@
+#include <string>
+
+#include "absl/status/status.h"
+#include "absl/status/statusor.h"
+
+#include "komfydb/common/tuple.h"
+#include "komfydb/common/tuple_desc.h"
+#include "komfydb/execution/op_iterator.h"
+#include "komfydb/transaction/transaction_id.h"
+
+namespace komfydb::execution {
+
+class SeqScan : OpIterator {
+ private:
+  TransactionId tid;
+  int table_id;
+  std::string table_alias;
+  DbFileIterator iterator;
+
+ public:
+  std::shared_ptr<Catalog> catalog;
+
+  SeqScan(TransactionId tid, int table_id, std::string table_alias)
+      : tid(tid), table_id(table_id), table_alias(table_alias) {}
+
+  SeqScan(TransactionId tid, int table_id) {
+    SeqScan(tid, table_id, common::GenerateUuidV4());
+  }
+
+  std::string GetTableName() {
+    ASSIGN_OR_RETURN(std::string table_name, catalog->GetTableName(table_id);
+    return table_name;
+  }
+
+  std::string GetAlias() { return table_alias; }
+
+  absl::Status Open() {
+    ASSIGN_OR_RETURN(DbFile * file, catalog->GetDatabaseFile(table_id));
+    iterator = file.Iterator();
```

Comment:
Didn't we establish that the Iterator shouldn't be taken from DbFile? IMHO SeqScan should never use DbFile directly. My idea:
- add a `DbFileIterator` (abstract? and derived HeapFileIterator? Mind future DbFiles for indexing) class
- Let it have a dependency on BufferPool and fetch pages directly from it.